rs/pocket_ic_server/tests/test.rs

```diff
@@ -685,3 +686,67 @@ fn canister_state_dir() {
     pic.update_call(nns_canister_id, Principal::anonymous(), "write", vec![])
         .unwrap();
 }
+
+/// Test that PocketIC can handle synchronous update calls, i.e. `/v3/.../call`.
+#[test]
+fn test_specified_id_call_v3() {
+    use ic_agent_call_v3::agent::CallResponse;
+
+    // Create live PocketIc instance.
+    let mut pic = PocketIcBuilder::new()
+        .with_nns_subnet()
+        .with_application_subnet()
+        .build();
+    let endpoint = pic.make_live(None);
+
+    // We define a "specified" canister ID that exists on the IC mainnet,
+    // but belongs to the canister ranges of no subnet on the PocketIC instance.
+    let specified_id = Principal::from_text("rimrc-piaaa-aaaao-aaljq-cai").unwrap();
+    assert!(pic.get_subnet(specified_id).is_none());
+
+    // We create a canister with that specified canister ID: this should succeed
+    // and a new subnet should be created.
+    let rt = tokio::runtime::Builder::new_current_thread()
+        .enable_all()
+        .build()
+        .unwrap();
+
+    // retrieve the first canister ID on the application subnet
+    // which will be the effective and expected canister ID for canister creation
+    let topology = pic.topology();
+
+    let app_subnet = topology.get_app_subnets()[0];
+    let effective_canister_id =
+        raw_canister_id_range_into(&topology.0.get(&app_subnet).unwrap().canister_ranges[0]).start;
+
+    rt.block_on(async {
+        let agent = ic_agent_call_v3::Agent::builder()
+            .with_url(endpoint.clone())
+            .build()
+            .unwrap();
+        agent.fetch_root_key().await.unwrap();
+
+        let arg = ProvisionalCreateCanisterWithCyclesArgs {
+            amount: None,
+            settings: None,
+            specified_id: None,
+            sender_canister_version: None,
+        };
+        let bytes = candid::Encode!(&arg).unwrap();
+
+        agent
```

Comment:
Could you please add a comment that we're actually testing /api/v3/.../call here?